File: content-solutions/hash/diff-pairs.java
import java.util.*;

class Solution {
    public long solution(int[] heights, int k) {
        Map<Integer, Integer> count = new HashMap<>();
        long answer = 0;
        for (int x : heights) {
            if (k == 0) answer += count.getOrDefault(x, 0);
            else answer += count.getOrDefault(x - k, 0) + count.getOrDefault(x + k, 0);
            count.put(x, count.getOrDefault(x, 0) + 1);
        }
        return answer;
    }
}

File: content-solutions/hash/diff-pairs.py
def solution(heights, k):
    count = {}
    answer = 0
    for x in heights:
        if k == 0:
            answer += count.get(x, 0)
        else:
            answer += count.get(x - k, 0) + count.get(x + k, 0)
        count[x] = count.get(x, 0) + 1
    return answer
